pandas/core/internals/blocks.py

```diff
@@ -1987,11 +1987,9 @@ class DatetimeTZBlock(DatetimeLikeBlock):
     _validate_ndim = True
     _can_consolidate = False
 
-    def values_for_json(self) -> np.ndarray:
-        # force dt64tz to go through object dtype
-        # tz info will be lost when converting to
-        # dt64 which is naive
-        return self.values.astype(object)
+    # Don't use values_for_json from DatetimeLikeBlock since it is
+    # an invalid optimization here(drop the tz)
+    values_for_json = NDArrayBackedExtensionBlock.values_for_json
```

Comment:
Does this fix a bug? From the comments I imagine this would produce different but better behavior that might be good to test.

Reply:
This was a followup to #46730. Behavior is unchanged, since the two are equivalent.